src/Dfe.PlanTech.Data.Sql/Entities/EstablishmentEntity.cs

```diff
@@ -7,53 +7,35 @@ namespace Dfe.PlanTech.Data.Sql.Entities;
 [Table("establishment")]
 public class EstablishmentEntity
 {
-    public const int EstablishmentRefLength = 50;
-    public const int EstablishmentTypeLength = 50;
-    public const int OrgNameLength = 200;
+    public const int EstablishmentRefMaxLengthInclusive = 50;
+    public const int EstablishmentTypeMaxLengthInclusive = 50;
+    public const int OrgNameMaxLengthInclusive = 200;
 
     private string _establishmentRef = null!;
     private string? _establishmentType;
     private string _orgName = null!;
 
     public int Id { get; set; }
 
-    [StringLength(EstablishmentRefLength)]
+    [StringLength(EstablishmentRefMaxLengthInclusive)]
     public string EstablishmentRef
     {
         get => _establishmentRef;
-        set
-        {
-            _establishmentRef = value.Length < EstablishmentRefLength
-                ? value
-                : value.AsSpan(0, EstablishmentRefLength).ToString();
-        }
+        set => _establishmentRef = TrimToMax(value, EstablishmentRefMaxLengthInclusive);
     }
 
-    [StringLength(EstablishmentTypeLength)]
+    [StringLength(EstablishmentTypeMaxLengthInclusive)]
     public string? EstablishmentType
     {
         get => _establishmentType;
-        set
-        {
-            if (value == null)
-                _establishmentType = null;
-            else
-                _establishmentType = value.Length < EstablishmentTypeLength
-                    ? value
-                    : value.AsSpan(0, EstablishmentTypeLength).ToString();
```

Comment:
Note here that we are _definitely_ doing an inclusive max length check here: if the input length is equal to the `const`'s max length, we trim to the max length (i.e., confirming that a `<=` is suitable here).